src/body.rs

```diff
@@ -117,6 +120,48 @@ fn mk_err<T>(_: T) -> Response {
     StatusCode::BAD_REQUEST.into_response()
 }
 
+pub struct MultipartFile(pub Vec<u8>);
+
+impl<S: 'static> Extract<S> for MultipartFile {
+    // Note: cannot use `existential type` here due to ICE
+    type Fut = FutureObj<'static, Result<Self, Response>>;
+
+    fn extract(data: &mut S, req: &mut Request, params: &RouteMatch<'_>) -> Self::Fut {
+        // https://stackoverflow.com/questions/43424982/how-to-parse-multipart-forms-using-abonander-multipart-with-rocket
+        let ct = req
+            .headers()
+            .get("content-type")
+            .expect("no header content-type");
```

Comment:
Note that `expect` will panic on  failure, which is probably not  what we want here -- I suspect we want to return an error response instead?

Reply:
Indeed, this is a highly WIP PR :) I would like to know whether I am pointing at the correct direction then later try fix all the details that can be fixed 😆

Reply:
hmmm... I am not sure how to deal with this option and return early here... also see my new commit and comments below. (Still working on it...)

Reply:
pushed one more commit 995bba8, but not sure if my way of early returning FutureObj with async is the best way to deal with it though 🤔